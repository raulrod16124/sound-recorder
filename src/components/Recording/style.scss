@import "./../../index.scss";
@import "react-h5-audio-player/src/styles.scss";

.recording {
  display: grid;
  grid-template-columns: 2fr;
  grid-template-rows: 2fr;
  row-gap: 0.5rem;
  width: 100%;
  margin-top: -1rem;

  .content-icons {
    width: 90%;
    display: flex;
    justify-content: space-between;
    margin: 1rem auto;
  }

  .shadow-icon-content {
    width: 6rem;
    height: 6rem;
    background-color: $primaryDeep;
    border-radius: 50%;

    &.active {
      background-color: rgba($color: $secondary, $alpha: 0.5);
    }

    .icon-content {
      position: relative;
      width: 80%;
      height: 80%;
      margin: 10% auto;
      background-color: $primaryDeep;
      box-shadow: inset -0.2rem 0.2rem 0.2rem $primary;
      border: 0.2rem solid $black;
      border-radius: 50%;
      display: flex;
      align-items: center;
      justify-content: center;

      &.active {
        background-color: $secondary;
        box-shadow: inset -0.2rem 0.2rem 0.2rem $secondaryLight;
      }

      & svg {
        padding-left: 0.3rem;
      }
    }
  }
}

.audio {
  display: flex;
  width: 90%;
  margin: 2rem auto 0 auto;
  border-radius: 1rem;
  background-color: $primary;
  color: $secondary;

  & div {
    border: none;
    color: $secondary;
  }

  & svg {
    color: $secondary;
  }
}

.rhap_container {
  border: none;
}
.vanish {
  animation: vanish 1s 1;
}

@keyframes vanish {
  0% {
    opacity: 1;
  }
  10% {
    opacity: 0.8;
  }
  20% {
    opacity: 0.6;
  }
  30% {
    opacity: 0.4;
  }
  60% {
    opacity: 0.2;
  }
  100% {
    opacity: 0;
  }
}
