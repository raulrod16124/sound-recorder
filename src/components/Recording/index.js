import "react-h5-audio-player/src/styles.scss";

import "./style.scss";

import PropTypes from "prop-types";
import AudioPlayer from "react-h5-audio-player";

import Icon from "../../stories/Icons";

const Recording = ({
  name,
  stream,
  handleOpenConfirmationPrompt,
  confirmationPromptState,
  qrState,
  setQrState,
}) => {
  // Class
  let className = require("classnames");
  let sendClass = className("icon-content qr-visibility", {
    active: qrState.visibility,
  });
  let shadowSendClass = className("shadow-icon-content", {
    active: qrState.visibility,
  });
  let trashClass = className("icon-content", {
    active: confirmationPromptState.openDelete,
  });
  let shadowTrashClass = className("shadow-icon-content", {
    active: confirmationPromptState.openDelete,
  });

  return (
    <>
      <article className="recording">
        <div className="content-icons">
          <div className={shadowSendClass}>
            <div
              className={sendClass}
              onClick={() => {
                setQrState({ visibility: true, url: stream });
              }}
            >
              <Icon
                icon="send"
                size="3rem"
                color={qrState.visibility ? "#373644" : "#28cac0"}
              />
            </div>
          </div>
          <div className={shadowTrashClass}>
            <div
              className={trashClass}
              onClick={() =>
                handleOpenConfirmationPrompt(
                  `Do you want to delete the recording ${name}?`,
                  "Delete"
                )
              }
            >
              <Icon
                icon="trash"
                size="3rem"
                color={
                  confirmationPromptState.openDelete ? "#373644" : "#28cac0"
                }
              />
            </div>
          </div>
        </div>
        <AudioPlayer
          className="audio"
          src={stream}
          autoPlayAfterSrcChange={false}
          customAdditionalControls={[]}
          showJumpControls={false}
          customVolumeControls={[]}
        />
      </article>
    </>
  );
};

Recording.defaultProps = {
  stream: {
    stream: "blob:http://localhost",
    name: "Default recording name",
    id: "id0",
  },
};

Recording.propTypes = {
  stream: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string,
  onDeleteHandler: PropTypes.func,
  onEditNameHandler: PropTypes.func,
};

export default Recording;
