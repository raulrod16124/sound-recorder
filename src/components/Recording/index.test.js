import { render, screen } from "@testing-library/react";
import React from "react";

import Recording from "./index";

describe("Testing the Recording Component", () => {
  let wrapper;

  // mocks
  let handleOpenConfirmationPrompt = jest.fn();
  let confirmationPromptState = { openDelete: false };
  let qrState = { visibility: false };
  let setQrState = jest.fn();

  const stream = {
    key: "lsdkjflds",
    stream: "sdkfjsdf08sdf",
    name: "2021-06-18 07:37:46",
  };

  beforeEach(() => {
    wrapper = render(
      <Recording
        stream={stream.stream}
        name={stream.name}
        key={stream.key}
        qrState={qrState}
        setQrState={setQrState}
        confirmationPromptState={confirmationPromptState}
        handleOpenConfirmationPrompt={handleOpenConfirmationPrompt}
      />
    );
  });

  it("should render the Recording component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should find the article role", () => {
    expect(screen.getByRole("article")).toBeInTheDocument();
  });

  it("should find the group role", () => {
    expect(screen.getByRole("group")).toBeInTheDocument();
  });

  it("should find the progressbar role", () => {
    expect(screen.getByRole("progressbar")).toBeInTheDocument();
  });

  it("should find the button role", () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  //   it("should..", () => {
  //     screen.getByRole("red");
  //   });

  //   it("renders without crashing", () => {
  //     const div = document.createElement("div");
  //     render(
  //       <Recording
  //         stream={stream.stream}
  //         name={stream.name}
  //         key={stream.key}
  //         qrState={qrState}
  //         setQrState={setQrState}
  //         confirmationPromptState={confirmationPromptState}
  //         handleOpenConfirmationPrompt={handleOpenConfirmationPrompt}
  //       />
  //     );
  //   });
});
