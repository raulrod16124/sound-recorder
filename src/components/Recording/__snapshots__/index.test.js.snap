// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Recording Component should render the Recording component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="recording"
      >
        <div
          class="content-icons"
        >
          <div
            class="shadow-icon-content"
          >
            <div
              class="icon-content qr-visibility"
            >
              <svg
                class="sc-bdvvtL kAkDOY"
                viewBox="0 0 95 95"
              >
                <path
                  class="sc-gsDKAQ iZbrEK"
                  clip-rule="evenodd"
                  d="M40 0C41.5151 0 42.9001 0.866708 43.5777 2.23878L79.5777 75.1386C80.2678 76.5361 80.1024 78.2108 79.1528 79.4423C78.2032 80.6738 76.6386 81.2423 75.1322 80.9034L40 72.9987L4.86775 80.9034C3.3614 81.2423 1.79684 80.6738 0.847208 79.4423C-0.102429 78.2108 -0.267771 76.5361 0.422321 75.1386L36.4223 2.23878C37.0999 0.866708 38.4849 0 40 0ZM44 65.6011L68.6639 71.1505L44 21.206V65.6011ZM36 21.206L11.3361 71.1505L36 65.6011V21.206Z"
                  fill-rule="evenodd"
                  stroke-linecap="round"
                  style="color: rgb(40, 202, 192);"
                />
              </svg>
            </div>
          </div>
          <div
            class="shadow-icon-content"
          >
            <div
              class="icon-content"
            >
              <svg
                class="sc-bdvvtL kAkDOY"
                viewBox="0 0 95 95"
              >
                <path
                  class="sc-gsDKAQ iZbrEK"
                  clip-rule="evenodd"
                  d="M21 8.4C21 3.76081 24.7608 0 29.4 0H54.6C59.2392 0 63 3.76081 63 8.4V16.8H71.3567C71.3826 16.7998 71.4086 16.7998 71.4347 16.8H79.8C82.1196 16.8 84 18.6804 84 21C84 23.3196 82.1196 25.2 79.8 25.2H75.3107L71.668 76.1985C71.354 80.5942 67.6963 84 63.2893 84H20.7107C16.3037 84 12.646 80.5942 12.332 76.1985L8.6893 25.2H4.2C1.8804 25.2 0 23.3196 0 21C0 18.6804 1.8804 16.8 4.2 16.8H12.5653C12.5914 16.7998 12.6174 16.7998 12.6433 16.8H21V8.4ZM29.4 16.8H54.6V8.4H29.4V16.8ZM17.1107 25.2L20.7107 75.6H63.2893L66.8893 25.2H17.1107Z"
                  fill-rule="evenodd"
                  stroke-linecap="round"
                  style="color: rgb(40, 202, 192);"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          aria-label="Audio player"
          class="rhap_container rhap_loop--off rhap_play-status--paused audio"
          role="group"
          tabindex="0"
        >
          <audio
            preload="auto"
            src="sdkfjsdf08sdf"
          />
          <div
            class="rhap_main rhap_stacked"
          >
            <div
              class="rhap_progress-section"
            >
              <div
                class="rhap_time rhap_current-time"
                id="rhap_current-time"
              >
                --:--
              </div>
              <div
                aria-label="Audio progress control"
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="0"
                class="rhap_progress-container"
                role="progressbar"
                tabindex="0"
              >
                <div
                  class="rhap_progress-bar rhap_progress-bar-show-download"
                >
                  <div
                    class="rhap_progress-indicator"
                    style="left: 0%;"
                  />
                  <div
                    class="rhap_progress-filled"
                    style="width: 0%;"
                  />
                </div>
              </div>
              <div
                class="rhap_time rhap_total-time"
              >
                --:--
              </div>
            </div>
            <div
              class="rhap_controls-section"
            >
              <div
                class="rhap_additional-controls"
              />
              <div
                class="rhap_main-controls"
              >
                <button
                  aria-label="Play"
                  class="rhap_button-clear rhap_main-controls-button rhap_play-pause-button"
                  type="button"
                >
                  <svg
                    focusable="false"
                    height="1em"
                    preserveAspectRatio="xMidYMid meet"
                    style="transform: rotate(360deg);"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 16.5v-9l6 4.5M12 2A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="rhap_volume-controls"
              />
            </div>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="recording"
    >
      <div
        class="content-icons"
      >
        <div
          class="shadow-icon-content"
        >
          <div
            class="icon-content qr-visibility"
          >
            <svg
              class="sc-bdvvtL kAkDOY"
              viewBox="0 0 95 95"
            >
              <path
                class="sc-gsDKAQ iZbrEK"
                clip-rule="evenodd"
                d="M40 0C41.5151 0 42.9001 0.866708 43.5777 2.23878L79.5777 75.1386C80.2678 76.5361 80.1024 78.2108 79.1528 79.4423C78.2032 80.6738 76.6386 81.2423 75.1322 80.9034L40 72.9987L4.86775 80.9034C3.3614 81.2423 1.79684 80.6738 0.847208 79.4423C-0.102429 78.2108 -0.267771 76.5361 0.422321 75.1386L36.4223 2.23878C37.0999 0.866708 38.4849 0 40 0ZM44 65.6011L68.6639 71.1505L44 21.206V65.6011ZM36 21.206L11.3361 71.1505L36 65.6011V21.206Z"
                fill-rule="evenodd"
                stroke-linecap="round"
                style="color: rgb(40, 202, 192);"
              />
            </svg>
          </div>
        </div>
        <div
          class="shadow-icon-content"
        >
          <div
            class="icon-content"
          >
            <svg
              class="sc-bdvvtL kAkDOY"
              viewBox="0 0 95 95"
            >
              <path
                class="sc-gsDKAQ iZbrEK"
                clip-rule="evenodd"
                d="M21 8.4C21 3.76081 24.7608 0 29.4 0H54.6C59.2392 0 63 3.76081 63 8.4V16.8H71.3567C71.3826 16.7998 71.4086 16.7998 71.4347 16.8H79.8C82.1196 16.8 84 18.6804 84 21C84 23.3196 82.1196 25.2 79.8 25.2H75.3107L71.668 76.1985C71.354 80.5942 67.6963 84 63.2893 84H20.7107C16.3037 84 12.646 80.5942 12.332 76.1985L8.6893 25.2H4.2C1.8804 25.2 0 23.3196 0 21C0 18.6804 1.8804 16.8 4.2 16.8H12.5653C12.5914 16.7998 12.6174 16.7998 12.6433 16.8H21V8.4ZM29.4 16.8H54.6V8.4H29.4V16.8ZM17.1107 25.2L20.7107 75.6H63.2893L66.8893 25.2H17.1107Z"
                fill-rule="evenodd"
                stroke-linecap="round"
                style="color: rgb(40, 202, 192);"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        aria-label="Audio player"
        class="rhap_container rhap_loop--off rhap_play-status--paused audio"
        role="group"
        tabindex="0"
      >
        <audio
          preload="auto"
          src="sdkfjsdf08sdf"
        />
        <div
          class="rhap_main rhap_stacked"
        >
          <div
            class="rhap_progress-section"
          >
            <div
              class="rhap_time rhap_current-time"
              id="rhap_current-time"
            >
              --:--
            </div>
            <div
              aria-label="Audio progress control"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="0"
              class="rhap_progress-container"
              role="progressbar"
              tabindex="0"
            >
              <div
                class="rhap_progress-bar rhap_progress-bar-show-download"
              >
                <div
                  class="rhap_progress-indicator"
                  style="left: 0%;"
                />
                <div
                  class="rhap_progress-filled"
                  style="width: 0%;"
                />
              </div>
            </div>
            <div
              class="rhap_time rhap_total-time"
            >
              --:--
            </div>
          </div>
          <div
            class="rhap_controls-section"
          >
            <div
              class="rhap_additional-controls"
            />
            <div
              class="rhap_main-controls"
            >
              <button
                aria-label="Play"
                class="rhap_button-clear rhap_main-controls-button rhap_play-pause-button"
                type="button"
              >
                <svg
                  focusable="false"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="transform: rotate(360deg);"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 16.5v-9l6 4.5M12 2A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
            <div
              class="rhap_volume-controls"
            />
          </div>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
